iiod/responder.c

```diff
@@ -681,6 +697,24 @@ static void handle_create_block(struct parser_pdata *pdata,
 	entry->io = io;
 	entry->client_id = cmd->client_id;
 
+	if (WITH_IIOD_USB_DMABUF && pdata->is_usb) {
+		entry->dmabuf_fd = iio_block_get_dmabuf_fd(block);
+		if (entry->dmabuf_fd > 0) {
+			ep_fd = buf_entry->is_tx ? pdata->fd_in : pdata->fd_out;
+
+			ret = usb_attach_dmabuf(ep_fd, entry->dmabuf_fd);
+			if (!ret) {
+				/* We could attach to functionfs. Disable CPU
+				 * access to the block as we won't need it. */
+				iio_block_disable_cpu_access(block, true);
+			} else {
+				/* If we can't attach - no problem. The
+				 * data will be transferred the regular way. */
+				entry->dmabuf_fd = -EINVAL;
```

Comment:
hmm, can't we use a better error code in here? "Invalid argument" is not really the same as "Functionality not supported.". Is there any subtleties that enforces other error code than "ENOSYS" (seems to be the proper fit)?

Reply:
The error code isn't really used in itself, any negative value will be treated as "zero-copy not possible". So I can change it to ENOSYS, it doesn't really matter.

Reply:
I guess at the very least matters in terms for reading the code... But if that's the case, stating that in the comment would be enough